### PR TITLE
temporary split script and fit script to be checked. After that I wil…

### PR DESCRIPTION
**Pull request by SimonCalo**
…l make them compatible for more users/ fit shapes

### CODE REVIEW

File: analysis/Scripts/SplitScript.py [hunk 1]
```diff
@@ -32,10 +39,10 @@
...
         mass_cuts = "lcplus_MM > 2375"
     for ybin in ybins:
-        for ptbin in ptbins:
-            yptcut = "lcplus_PT >= {0} && lcplus_PT < {1} && lcplus_RAPIDITY >= {2} && lcplus_RAPIDITY < {3}".format(ptbin[0], ptbin[1], ybin[0], ybin[1])
+        for pbin in pbins:
+            yptcut = "lcplus_PT >= {0} && lcplus_PT < {1} && lcplus_RAPIDITY >= {2} && lcplus_RAPIDITY < {3}".format(pbin[0], pbin[1], ybin[0], ybin[1])
             allcuts = "{0} && {1} && {2}".format(globalCuts, yptcut, mass_cuts)
-            wfile = TFile.Open(directory + particle + "_splitfile_y{0}-{1}_pt{2}-{3}.root".format(ybin[0],ybin[1],ptbin[0],ptbin[1]), "RECREATE")
+            wfile = TFile.Open(directory + particle + "_splitfile_y{0}-{1}_p{2}-{3}.root".format(ybin[0],ybin[1],pbin[0],pbin[1]), "RECREATE")
             subtree = alldata.CopyTree( allcuts )
```
Comment (jackfrost373) [inline analysis/Scripts/SplitScript.py]: The issue here is that you get two copies of the DecayTree in your new TFile, correct? And they don't seem to have the same Nentries (i.e. a copy _before_ the cut, and one after, is written). That's a nuisance, and a wonderful example of crappy memory management. Could you try to swap the order of opening the TFile and declaring the new TTree? i.e. 

```suggestion
            subtree = alldata.CopyTree( allcuts )
            wfile = TFile.Open(directory + particle + "_splitfile_y{0}-{1}_p{2}-{3}.root".format(ybin[0],ybin[1],pbin[0],pbin[1]), "RECREATE")
            wfile.cd()
            subtree.Write()
```

If that does not seem to work, try opening another file to write to before the loop, where ROOT might dump it's trees without asking, and then opening our new file etc. in the loop, just like you do here?

Comment (jackfrost373) [reply analysis/Scripts/SplitScript.py]: As I said, I checked the effect (it's a discussion within the b2mumu group as well), and it's all OK. Just don't call it with the number, just the name 'DecayTree'.

**Comment by jackfrost373** [comment]
One more suggestion I have: we can ideally put 'helper' scripts into a folder called 'python', ready to be included. We can put 'cross-check' or simple one-use things into a folder called 'scripts'. That way, only the main scripts (i.e. "do_fit" or "plot_comparisons") are in the main folder - which makes the file/folder structure very clear.

File: analysis/Scripts/FitScript.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+ptbins = [ [3000,4000],[4000,5000], [5000,6000], [6000,7000], [7000,8000], [8000,10000], [10000,15000]] # Transverse momentum bins
+
+text = open(directory + "Output.txt", "w+") # This is the text file in which all of the interesting parameters from the fit are printed
+
+# loop used to fit all the bins and print all of the interesting parameters of the fit
+for particle in particles:
+    for ybin in ybins:
+        for ptbin in ptbins:
+            location = (directory + particle + "_splitfile_y{0}-{1}_pt{2}-{3}.root".format(ybin[0],ybin[1],ptbin[0],ptbin[1]))
```
[the remark this answers is not in the document]
Comment (jackfrost373) [reply analysis/Scripts/FitScript.py]: check

File: analysis/Scripts/FitScript.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import ROOT, os, Data_Bukin_Mass_Fit_function
+from ROOT import TChain, TFile
+from Data_Bukin_Mass_Fit_function import Bukin_fit
```
Comment (SimonCalo) [inline analysis/Scripts/FitScript.py]: check

**Comment by jackfrost373** [comment]
Hi @SimonCalo @cpawley , what's the status on these comments? Did you have a go, or do we choose to ignore them? In any case, if you can resolve my comments, we can merge it ;-)

**Comment by SimonCalo** [comment]
Hi @jackfrost373, I was thinking about it and we can add it if you think that it would be useful, but personally I do not see too much added value into doing that since I think the splitfile name we are only going to use in this specific instance and thus adding it to the general imports module for being used in only one instance does not seem too necessary to me. That said, if you guys think that it would help, then I will be happy to add that specific function today :)

**Comment by jackfrost373** [comment]
Wow, that's a long sentence @SimonCalo :) 
OK, then I resolve all points and merge.

**Pull request merged**